server/src/signin-button.js

```diff
@@ -2,44 +2,69 @@ const React = require("react");
 const PropTypes = require("prop-types");
 const { Localized } = require("fluent-react/compat");
 const sendEvent = require("./browser-send-event.js");
+const { FxaOnboardingDialog } = require("./fxa-onboarding-dialog");
 
 exports.SignInButton = class SignInButton extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
       displaySettings: props.isFxaAuthenticated,
+      hasFxaOnboardingDialog: props.hasFxaOnboardingDialog,
     };
   }
 
   static getDerivedStateFromProps(nextProps, prevState) {
-    return { displaySettings: nextProps.isFxaAuthenticated };
+    return { displaySettings: nextProps.isFxaAuthenticated};
+  }
+
+  componentDidUpdate(oldProps, oldState) {
+    if (oldProps.hasFxaOnboardingDialog !== this.props.hasFxaOnboardingDialog) {
+      this.setState({hasFxaOnboardingDialog: this.props.hasFxaOnboardingDialog});
+    }
   }
 
   render() {
     if (this.state.displaySettings) {
-      return <Localized id="buttonSettings" attrs={{title: true}}>
-          <a className="transparent nav-button icon-settings" tabIndex="0" href="/settings" title="Settings">
-            <img src={this.props.staticLink("/static/img/icon-settings.svg")} />
-          </a>
-        </Localized>;
+      return <div className="fxa-signin">
+          <Localized id="buttonSettings" attrs={{title: true}}>
+            <a className="transparent nav-button icon-settings" tabIndex="0" href="/settings" title="Settings">
+              <img src={this.props.staticLink("/static/img/icon-settings.svg")} />
+            </a>
+          </Localized>
+        </div>;
     }
 
     const logInURI = "/api/fxa-oauth/login/" + this.props.initialPage;
-    return <Localized id="buttonSignIn" attrs={{title: true}}>
-        <a className="transparent nav-button icon-settings" tabIndex="0" href={logInURI} title="SignIn"
-           onClick={this.clickHandler.bind(this)}>
-          <img src={this.props.staticLink("/static/img/icon-settings.svg")} />
-        </a>
-      </Localized>;
+    return <div className="fxa-signin">
+        <Localized id="buttonSignIn" attrs={{title: true}}>
+          <a className="transparent nav-button icon-settings" tabIndex="0" href={logInURI} title="SignIn"
+             onClick={this.clickHandler.bind(this)}>
+            <img src={this.props.staticLink("/static/img/icon-settings.svg")} />
+          </a>
+        </Localized>
+        <FxaOnboardingDialog logInURI={logInURI} display={this.state.hasFxaOnboardingDialog}
```

Comment:
Remove the `display` prop from the FxaOnboardingDialog and just conditionally render it here

Reply:
+1, Updated with feedback. Thanks!